zipline/sources/data_frame_source.py

```diff
@@ -111,46 +112,43 @@ def __init__(self, data, **kwargs):
 
         # Hash_value for downstream sorting.
         self.arg_string = hash_args(data, **kwargs)
-
-        self._raw_data = None
-
-    @property
-    def mapping(self):
-        mapping = {
-            'dt': (lambda x: x, 'dt'),
-            'sid': (lambda x: x, 'sid'),
-            'price': (float, 'price'),
-            'volume': (int, 'volume'),
-        }
-
-        # Add additional fields.
-        for field_name in self.data.minor_axis:
-            if field_name in ['price', 'volume', 'dt', 'sid']:
-                continue
-            mapping[field_name] = (lambda x: x, field_name)
-
-        return mapping
+        self.it = self.mapped_data
 
     @property
     def instance_hash(self):
         return self.arg_string
 
-    def raw_data_gen(self):
-        for dt in self.data.major_axis:
-            df = self.data.major_xs(dt)
-            for sid, series in df.iteritems():
-                if sid in self.sids:
-                    event = {
-                        'dt': dt,
-                        'sid': sid,
-                    }
-                    for field_name, value in series.iteritems():
-                        event[field_name] = value
-
-                    yield event
+    def raw_data(self):
+        # treat yoself
+        pass
 
     @property
-    def raw_data(self):
-        if not self._raw_data:
-            self._raw_data = self.raw_data_gen()
-        return self._raw_data
+    def mapped_data(self):
+        # currently volume is same dtype as prices, float64
```

Comment:
It's probably better, or at least ok, to have the volumes as float64.

For places where volume needs to be multiplied by the price, it should be faster to do the float*float.

Also, allows for shaping data in both directions, i.e. ohlcv can be a vector... if we care about that.